pkg/tfbridge/names.go

```diff
@@ -66,6 +75,17 @@ func PulumiToTerraformName(name string, tfs shim.SchemaMap, ps map[string]*Schem
 	return result
 }
 
+// A strict inverse of TerraformToPulumiName(upper=false) that tabulates it based on the SchemaMap.
+func pulumiToTerraformNameTable(schemaMap shim.SchemaMap, ps map[string]*SchemaInfo) map[string]string {
```

Comment:
Calling this as part of PulumiToTerraformName when that loops over all properties is quadratic in properties, which maybe is bad if we have 1000-property situations. There is a way to mitigate this inside the tfbridge though with a bit of refactoring. Let me link another experimental PR shortly.